markup_trace.py

```diff
@@ -95,10 +95,13 @@ def lookup_srcloc(self, srcloc):
 class Trace:
     """Trace annotated with debugging information."""
 
-    def __init__(self, name, trace, symbols, properties, snippets, outdir='.'):
+    def __init__(self, name, trace, symbols, properties, loops, snippets,
+                 outdir='.'):
         self.prop_name = name
         self.prop_desc = properties.get_description(name)
-        self.prop_srcloc = format_srcloc(properties.get_srcloc(name), symbols)
+        self.prop_srcloc = format_srcloc(properties.get_srcloc(name)
+                                         or loops.lookup_assertion(name),
```

Comment:
What is going on here?

Reply:
Added comment: 

        # Having extracted the loop name from the assertion name, look
        # up the loop trying both its proper name and its mangled
        # name.